engine/language-server/src/main/scala/org/enso/languageserver/search/SuggestionsHandler.scala

```diff
@@ -362,7 +370,43 @@ final class SuggestionsHandler(
                 .map(CompletionResult.tupled)
           )
         }
-        .pipeTo(sender())
+        val checkForFirstTimeJava = res.andThen(v => {
+          logger.warn("resolved to " + v)
+          if (v.isSuccess && v.get.asInstanceOf[CompletionResult].results.isEmpty && isJavaClass) {
+            logger.warn("  empty result isn't enough, recomputing")
+            val c = Class.forName(allSelfTypes.head)
+            val newSuggestions = new java.util.ArrayList[Suggestion]
+            val typ = new Suggestion.Type(None, "Java", c.getName(), Nil, "", None, None, None)
+            newSuggestions.add(typ)
+            val names = new java.util.HashSet[String]
+            c.getMethods().foreach(m => {
+              val isNewName = names.add(m.getName())
+              if (isNewName && (m.getModifiers & java.lang.reflect.Modifier.STATIC) == 0 && m.getDeclaringClass() != classOf[java.lang.Object]) {
+                var cnt = 0;
+                val args = m.getParameterTypes().map(p => {
+                  cnt = cnt + 1
+                  new Suggestion.Argument("arg" + cnt, p.getName(), false, false, None, None)
+                }).toIndexedSeq;
+                val method = new Suggestion.DefinedMethod(None, "Java", m.getName(), args, c.getName, m.getReturnType().getName(), false, None, Nil)
+                newSuggestions.add(method)
+              }
+            })
+            logger.warn("Computed new replacement: " + newSuggestions)
+            applyLoadedSuggestions(newSuggestions.toArray(new Array[Suggestion](0)).toVector).andThen(r =>
+              if (r.isFailure) {
+                logger.error("database update failed: " + r)
+              } else {
+                logger.warn("database updated with: " + r)
```

Comment:
Here I got a bit lost, @4e6 - do you know how to convert these newly inserted suggestions into something to `pipeTo(sender())`? Obviously `Future(r.get).pipeTo(sender())` isn't correct...

Reply:
`sender()` method returns the sender actor of the currently processed message. You should not evaluate it inside of the lambda, but you can compute it outside, and then use it as closure, i.e.
```scala
val replyTo = sender()
val checkForFirstTimeJava = res.andThen(v => {
  ...
  replyTo ! message
})
```